juju/juju
# fix(cmr): expose all controller addresses cross-controller

We were previously sourcing addresses for cross-controller info from `APIHostPortsForAgents` and prioritising for local-cloud scope.

This meant that we were not exposing public addresses, resulting in some CMRs not being able to contact remote controllers.

Here we user `APIHostPortsForClients`, which includes public addresses, and we prioritise for the public scope.

## QA steps

Establish a CMR between 2 AWS controllers in different regions, which tests that the public controller IPs are exposed.
- `juju bootstrap aws/eu-west-1 src --debug --build-agent`
- `juju add-model mover`
- `juju deploy ./testcharms/charms/dummy-source --config token=INITIAL`
- `juju offer dummy-source:sink`
- `juju bootstrap aws/eu-west-2 cns --debug --build-agent`
- `juju add-model consumer`
- `juju consume src:admin/mover.dummy-source`
- `juju deploy ./testcharms/charms/dummy-sink`
- `juju relate dummy-source dummy-sink`
- Juju status should indicate successful establishment of the CMR:
```
Model     Controller  Cloud/Region   Version  SLA          Timestamp
consumer  cns         aws/eu-west-2  3.4.5.1  unsupported  12:09:57+02:00

SAAS          Status  Store  URL
dummy-source  active  src    admin/mover.dummy-source

App         Version  Status  Scale  Charm       Channel  Rev  Exposed  Message
dummy-sink           active      1  dummy-sink             0  no       Token is INITIAL

Unit           Workload  Agent  Machine  Public address  Ports  Message
dummy-sink/0*  active    idle   0        35.177.214.52          Token is INITIAL

Machine  State    Address        Inst id              Base          AZ          Message
0        started  35.177.214.52  i-0e7629e5133c0faef  ubuntu@22.04  eu-west-2a  running

Integration provider  Requirer           Interface    Type     Message
dummy-sink:source     dummy-source:sink  dummy-token  regular
```

## Documentation changes

None.

## Links

**Jira card:** [JUJU-6195](https://warthogs.atlassian.net/browse/JUJU-6195)



[JUJU-6195]: https://warthogs.atlassian.net/browse/JUJU-6195?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
